+bids/layout.m

```diff
@@ -187,11 +187,13 @@
     for i=1:numel(f)
         
         p = parse_filename(f{i}, {'sub','ses','task','acq','rec','fa','echo','inv','run','recording', 'meta'});
-        subject.func = [subject.func p];
-        subject.func(end).meta = bids.util.tsvread(fullfile(pth,f{i})); % ?
-
-    end
+        base = p.filename(1:(strfind(p.filename, ['_' p.type])-1));
```

Comment:
OK I got this is because the p.type is not appended that the `event` type is then lost and the test fails when we query for it.

Reply:
@robertoostenveld I suspect that the PR #17 might also pass after this is settled.

Reply:
Currently testing on BIDS-examples/ds000117:

- 'coordsystem' 
- 'events'
- 'headshape'

are also not returned anymore as types either, except for `events` refers to a purely behavioral experiment (`beh`).

Other possibility to fix this: we change the test so that we don't have those returned as types in the dataset. @gllmflndn what say you?

Reply:
I consider json/coordsystem/bvec/bval/events(*)/electrodes/headshape etc. as auxiliary files which enrich the core data (i.e. the brain data from the EEG amplifier or the MR scanner). Each of these belongs to an actual brain-data recording. That is why I (hopefully in line with BIDS specification) do not consider these as data `types`.

*) The only exception is that an `events.tsv` file in a `beh` is really the actual recording, since in that case there is no brain data.    

I have to admit that I did not look into the included CI test, I only ran manual tests on the bids-examples repository.

Reply:
Yup I am totally fine with that approach actually.